### ISSUE TITLE
Compilation warning with #define U8GLIB_SSD1306

### ISSUE THREAD

**Issue by ghost** [MarlinFirmware/Marlin#4614]
When `#define U8GLIB_SSD1306` is enabled with newest RCBugFix, it occurs compilation error.

<details>
<summary>

error message</summary>

```
AppData\Local\Temp\build6674025c445f92d0b07185e4e4080804.tmp\sketch\ultralcd.cpp: In function 'void lcd_update()':

ultralcd.cpp:2699: error: 'currentScreen' was not declared in this scope

           (*currentScreen)();

             ^
AppData\Local\Temp\build6674025c445f92d0b07185e4e4080804.tmp\sketch\ultralcd.cpp:426:13: warning: 'void lcd_status_screen()' defined but not used [-Wunused-function]

 static void lcd_status_screen() {

             ^

exit status 1
'currentScreen' was not declared in this scope
```

</details>

**Comment by thinkyhead** [comment]
I think this might be the right way to fix it…

``` diff
#if ENABLED(DOGLCD)  // Changes due to different driver architecture of the DOGM display
  static int8_t dot_color = 0;
  dot_color = 1 - dot_color;
  u8g.firstPage();
  do {
    lcd_setFont(FONT_MENU);
    u8g.setPrintPos(125, 0);
    u8g.setColorIndex(dot_color); // Set color for the alive dot
    u8g.drawPixel(127, 63); // draw alive dot
    u8g.setColorIndex(1); // black on white
-   (*currentScreen)();
+   #if ENABLED(ULTIPANEL)
+     (*currentScreen)();
+   #else
+     lcd_status_screen();
+   #endif
  } while (u8g.nextPage());
#elif ENABLED(ULTIPANEL)
  (*currentScreen)();
#else
  lcd_status_screen();
#endif
```

**Comment by ghost** [comment]
I got new compilation warnings.
<details>
<summary>

new warning message</summary>

 

```
AppData\Local\Temp\build6674025c445f92d0b07185e4e4080804.tmp\sketch\ultralcd_impl_DOGM.h:513:13: warning: 'lcd_implementation_drawmenu_generic' defined but not used [-Wunused-function]

 static void lcd_implementation_drawmenu_generic(bool isSelected, uint8_t row, const char* pstr, char pre_char, char post_char) {

             ^

AppData\Local\Temp\build6674025c445f92d0b07185e4e4080804.tmp\sketch\ultralcd_impl_DOGM.h:531:13: warning: '_drawmenu_setting_edit_generic' defined but not used [-Wunused-function]

 static void _drawmenu_setting_edit_generic(bool isSelected, uint8_t row, const char* pstr, const char* data, bool pgm) {

             ^
```

</details>

I'm not sure that these are problem or not, but compilation error and warning in my previous comment is gone.
Thanks for the fixing.

**Comment by thinkyhead** [comment]
It looks like they just aren't used if `ULTIPANEL` isn't set. So it's another patch I need to make.

**Comment by zenmetsu** [comment]
XD  @thinkyhead i'd kiss ya if I could.  I am working on adding my SSD1351 SPI OLED to the code and ran right into this issue.  Your suggested code change came just seconds after I needed it.  It works brilliantly.

**Comment by boelle** [comment]
@esenapaj does it now compile without error?

**Comment by ghost** [comment]
Yes it is, compilation error was fixed by commit 132322e4d94cf776ea3d42af68406bf7b012120a.
But new compilation warning was occured, and it looks like that @thinkyhead work in preparation for another patch,
therefore I reopened this Issue.

**Comment by boelle** [comment]
maybe close this one and open a new on with the new warning?
